packages/authorization-process/src/services/readModelService.ts

```diff
@@ -46,6 +59,92 @@ export function readModelServiceBuilder(
     ): Promise<WithMetadata<Client> | undefined> {
       return getClient({ "data.id": id });
     },
+
+    async getClients(
+      filters: GetClientsFilters,
+      { offset, limit }: { offset: number; limit: number }
+    ): Promise<ListResult<Client>> {
+      const { name, userIds, consumerId, purposeId, kind } = filters;
+
+      const nameFilter: ReadModelFilter<Client> = name
+        ? {
+            "data.name": {
+              $regex: ReadModelRepository.escapeRegExp(name),
+              $options: "i",
+            },
+          }
+        : {};
+
+      const userIdsFilter: ReadModelFilter<Client> =
+        ReadModelRepository.arrayToFilter(userIds, {
+          $or: userIds.map((userId) => ({ "data.users": { $eq: userId } })),
+        });
+
+      const consumerIdFilter: ReadModelFilter<Client> = {
+        "data.consumerId": { $eq: consumerId },
+      };
+
+      const purposeIdFilter: ReadModelFilter<Client> = purposeId
+        ? {
+            "data.purposes": { $eq: purposeId },
+          }
+        : {};
+
+      const kindFilter: ReadModelFilter<Client> = kind
+        ? {
+            "data.kind": {
+              $regex: ReadModelRepository.escapeRegExp(kind),
+              $options: "i",
+            },
```

Comment:
```suggestion
            "data.kind": { $eq: kind },
```
We don't need to do a case insensitive search for this, we can just check if it's equal